### PR TITLE
Feature/improve

### PR DESCRIPTION
**Pull request by thomasglachant**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /nbproject/*
-.idea/
\ No newline at end of file
+.idea/
+vendor/
```
Comment (nicoosh) [inline .gitignore]: missing blank line at the end of file.

File: Controller/StatusController.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Stadline\StatusPageBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class StatusController extends Controller
```
Comment (nicolasheraly) [inline Controller/StatusController.php]: Missing PHPDoc on class definition.

File: Controller/StatusController.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * @param Request $request
```
Comment (nicoosh) [inline Controller/StatusController.php:14]: Missing method description.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 4]
```diff
@@ -6,27 +6,30 @@
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Reference;
 
-class RequirementCompilerPass implements CompilerPassInterface 
+class RequirementCompilerPass implements CompilerPassInterface
```
Comment (nicolasheraly) [inline DependencyInjection/Compiler/RequirementCompilerPass.php]: Missing PHPDoc at class definition.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 5]
```diff
@@ -6,27 +6,30 @@
...
-		
-		foreach ($taggedServices as $id => $attributes) {
-			$definition->addMethodCall(
-				'addCollection',
-				array(new Reference($id))
-			);
-		}
-	}
+    /**
+     * @param ContainerBuilder $container
```
Comment (nicolasheraly) [inline DependencyInjection/Compiler/RequirementCompilerPass.php]: Missing method description.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 6]
```diff
@@ -6,27 +6,30 @@
...
-		);
-		
-		foreach ($taggedServices as $id => $attributes) {
-			$definition->addMethodCall(
-				'addCollection',
-				array(new Reference($id))
-			);
-		}
-	}
+    /**
```
Comment (nicoosh) [inline DependencyInjection/Compiler/RequirementCompilerPass.php:11]: Missing method description.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 7]
```diff
@@ -6,27 +6,30 @@
...
+        );
+
+        foreach ($taggedServices as $id => $attributes) {
+            $definition->addMethodCall(
+                'addCollection',
+                array(new Reference($id))
+            );
+        }
+    }
 }
```
Comment (nicoosh) [inline DependencyInjection/Compiler/RequirementCompilerPass.php]: Missing blank line at the end of file.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 8]
```diff
@@ -6,27 +6,30 @@
...
+    /**
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition("stadline_status_page.requirement.collections")) {
+            return;
+        }
+
+        $definition = $container->getDefinition(
```
Comment (nicolasheraly) [inline DependencyInjection/Compiler/RequirementCompilerPass.php]: Can be on single line.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 9]
```diff
@@ -6,27 +6,30 @@
...
-	}
+    /**
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition("stadline_status_page.requirement.collections")) {
+            return;
+        }
+
```
Comment (nicoosh) [inline DependencyInjection/Compiler/RequirementCompilerPass.php:21]: Can be on single line.

File: DependencyInjection/Compiler/RequirementCompilerPass.php [hunk 10]
```diff
@@ -6,27 +6,30 @@
...
+        $definition = $container->getDefinition(
+            'stadline_status_page.requirement.collections'
+        );
+
```
Comment (nicoosh) [inline DependencyInjection/Compiler/RequirementCompilerPass.php:23]: Can be on single line.

File: Requirements/RequirementCollectionInterface.php [hunk 11]
```diff
@@ -0,0 +1,58 @@
+<?php 
+
+namespace Stadline\StatusPageBundle\Requirements;
+
```
Comment (nicolasheraly) [inline Requirements/RequirementCollectionInterface.php]: Missing PHPDoc on interface definition.

File: Requirements/RequirementCollectionInterface.php [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+interface RequirementCollectionInterface
+{
+    /**
+     * Returns the name of requirements collection
```
Comment (nicoosh) [inline Requirements/RequirementCollectionInterface.php:8]: Missing a blank line between method description and @param

File: Requirements/RequirementCollectionInterface.php [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+     * @return string
+     */
+    public function getName();
+
+    /**
```
Comment (nicoosh) [inline Requirements/RequirementCollectionInterface.php:14]: Missing method description.

File: Requirements/RequirementCollectionInterface.php [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+     * Adds an optional recommendation.
+     *
+     * @param bool        $fulfilled   Whether the recommendation is fulfilled
+     * @param string      $testMessage The message for testing the recommendation
+     * @param string      $helpHtml    The help text formatted in HTML for resolving the problem
+     * @param string|null $helpText    The help text (when null, it will be inferred from $helpHtml, i.e. stripped from HTML tags)
+     */
+    public function addRecommendation($fulfilled, $testMessage, $helpHtml, $helpText = null);
+}
```
Comment (nicoosh) [inline Requirements/RequirementCollectionInterface.php]: Missing blank line at the end of file.

File: Requirements/RequirementCollections.php [hunk 15]
```diff
@@ -4,9 +4,15 @@
...
 class RequirementCollections implements \IteratorAggregate
 {
+    /**
+     * @var RequirementCollectionInterface[]
+     */
 	private $collections = array();
-	
-	public function addCollection($collection)
+
+    /**
```
Comment (nicoosh) [inline Requirements/RequirementCollections.php:12]: Missing method description.

File: Requirements/RequirementCollections.php [hunk 16]
```diff
@@ -61,48 +63,58 @@ public function getFailedRecommendations()
         return $array;
     }
 
+    /**
```
Comment (nicoosh) [inline Requirements/RequirementCollections.php:68]: Missing method description.

File: Requirements/RequirementCollections.php [hunk 17]
```diff
@@ -61,48 +63,58 @@ public function getFailedRecommendations()
...
             return true;
-        }
-        else
-        {
+        } else {
             return false;
         }
     }
-    
+
```
Comment (nicoosh) [inline Requirements/RequirementCollections.php:82]: Missing method description.

File: Requirements/RequirementCollections.php [hunk 18]
```diff
@@ -61,48 +63,58 @@ public function getFailedRecommendations()
...
-        foreach ($this->collections as $collection)
-        {
+        foreach ($this->collections as $collection) {
             $count += count($collection->getRequirements());
         }
         
         return $count;
     }
 
+    /**
```
Comment (nicoosh) [inline Requirements/RequirementCollections.php:99]: Missing method description.

File: Requirements/RequirementCollections.php [hunk 19]
```diff
@@ -61,48 +63,58 @@ public function getFailedRecommendations()
...
-        foreach ($this->collections as $collection)
-        {
+        foreach ($this->collections as $collection) {
             $count += count($collection->getRecommendations());
         }
     
         return $count;
     }
-    
+
```
Comment (nicoosh) [inline Requirements/RequirementCollections.php:114]: Missing method description.

File: Requirements/RequirementCollections.php [hunk 20]
```diff
@@ -61,48 +63,58 @@ public function getFailedRecommendations()
...
+    /**
+     * @return int
+     */
     public function countFailedRequirements()
     {
         return count($this->getFailedRequirements(), COUNT_RECURSIVE) - count($this->getFailedRequirements());
     }
 
+    /**
```
Comment (nicoosh) [inline Requirements/RequirementCollections.php:125]: Missing method description.

File: Requirements/SymfonyRequirements.php [hunk 21]
```diff
@@ -2,12 +2,13 @@
 
 namespace Stadline\StatusPageBundle\Requirements;
 
-require_once '../app/SymfonyRequirements.php';
-
-class SymfonyRequirements extends \SymfonyRequirements
+class SymfonyRequirements extends \SymfonyRequirements implements RequirementCollectionInterface
 {
+    /**
```
Comment (nicoosh) [inline Requirements/SymfonyRequirements.php:7]: Missing method description.

File: Requirements/VersionRequirements.php [hunk 22]
```diff
@@ -25,6 +28,9 @@ public function __construct($container)
         $this->addRequirement(isset($branch), "Git Branch", isset($branch) ? $branch : 'NONE');
     }
 
+    /**
```
Comment (nicoosh) [inline Requirements/VersionRequirements.php:31]: Missing method description.

File: Resources/config/routing.yml [hunk 23]
```diff
@@ -1,7 +1,3 @@
 stadline_status_page_homepage:
     pattern:  /status
-    defaults: { _controller: StadlineStatusPageBundle:Default:index}
-    
-stadline_status_page_params:
-    pattern:  /status/params
-    defaults: { _controller: StadlineStatusPageBundle:Default:exposeParams }
+    defaults: { _controller: StadlineStatusPageBundle:Status:index}
```
Comment (nicoosh) [inline Resources/config/routing.yml]: Missing blank line at the end of file.

File: Resources/config/services.yml [hunk 24]
```diff
@@ -1,32 +1,19 @@
...
-        class: %stadline_status_page.api.requirement.class%
-        arguments: ['@service_container']
-        tags:
-          - { name: status_page.requirement}
-
-    stadline_status_page.api.factory:
-        class: %stadline_status_page.api.factory.class%
-        calls:
-            - [setClient, ['@guzzle.client']]
+          - { name: "status_page.requirement" }
```
Comment (nicoosh) [inline Resources/config/services.yml]: Missing blank line at the end of file.

File: Resources/config/services.yml [hunk 25]
```diff
@@ -1,32 +1,19 @@
...
-    stadline_status_page.api.requirement.class: Stadline\StatusPageBundle\Requirements\ApiRequirements
-    stadline_status_page.api.factory.class: Stadline\StatusPageBundle\ApiStatus\ApiStatusFactory
+    stadline_status_page.symfony.requirement.class: "Stadline\StatusPageBundle\Requirements\SymfonyRequirements"
+    stadline_status_page.requirement.collections.class: "Stadline\StatusPageBundle\Requirements\RequirementCollections"
+    stadline_status_page.version.requirement.class: "Stadline\StatusPageBundle\Requirements\VersionRequirements"
 
 services:
     stadline_status_page.version.requirement:
-        class: %stadline_status_page.version.requirement.class%
+        class: "%stadline_status_page.version.requirement.class%"
```
Comment (nicoosh) [inline Resources/config/services.yml]: Be careful, with phpunit/phpunit > 5, all YAML files should contain only simple quotes.
I had some troubles in decathloncoach with it.

see https://github.com/stadline/mygeonaute2/commit/15acf74a95bd9ef5737a7297402a1dc7e7a3f18c

**Pull request closed without merging**